### PR TITLE
Use image_url() when creating the $logo_url variable

### LINKED ISSUE #1573: Don't use `image_url` in the default styles
https://github.com/projectblacklight/blacklight/blob/e7d7cb883e4d24a5263797cac47a8e5fd5b603ca/app/assets/stylesheets/blacklight/_header.scss#L6

this prevents us from using an in image not on the site.

### PR DESCRIPTION
**Pull request by cbeer**
Fixes #1573

### CODE REVIEW

**Comment by jcoyne** [comment]
This looks good, will it create a backward compatibility issue we should record somewhere?

**Comment by cbeer** [comment]
I hope a notice in the release notes ought to be sufficient.

**Comment by jcoyne** [comment]
Okay. Are you drafting those?

File: app/assets/stylesheets/blacklight/blacklight_defaults.scss [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /* Warning! If you want to change these, just copy them into your own theme css. But you want to remove the !default, which only will set them if not already set. */
 
-$logo_image: 'blacklight/logo.png' !default;
+$logo_image: image_url('blacklight/logo.png') !default;
```
Comment (jcoyne) [inline app/assets/stylesheets/blacklight/blacklight_defaults.scss]: Can we chang the variable to use hyphens?

**Comment by cbeer** [comment]
Ok, now using `$logo-url` with backwards compatible support for `$logo_url`.

**Pull request merged**